File: Colage/Shared/Discovery/FullProfileView.swift
import SwiftUI

/// Full profile view — expanded from mini sheet or list tap
struct FullProfileView: View {
    let student: NearbyStudent
    @Environment(\.dismiss) private var dismiss

    var body: some View {
        ZStack {
            ColageColors.background.ignoresSafeArea()

            ScrollView {
                VStack(spacing: 24) {
                    // Header with photo
                    VStack(spacing: 16) {
                        AvatarView(
                            imageURL: student.profile.profilePhotoURL,
                            size: 120,
                            initials: student.profile.displayName.initials
                        )

                        Text(student.profile.displayName)
                            .font(ColageFonts.title)
                            .foregroundStyle(ColageColors.textPrimary)

                        HStack(spacing: 12) {
                            // University badge
                            HStack(spacing: 4) {
                                Image(systemName: "graduationcap.fill")
                                    .font(.system(size: 12))
                                Text(student.profile.universityDomain.replacingOccurrences(of: ".edu", with: "").uppercased())
                                    .font(ColageFonts.captionBold)
                            }
                            .foregroundStyle(ColageColors.primary)
                            .padding(.horizontal, 10)
                            .padding(.vertical, 5)
                            .background(ColageColors.primary.opacity(0.12))
                            .clipShape(Capsule())

                            // Distance
                            HStack(spacing: 4) {
                                Image(systemName: "location.fill")
                                    .font(.system(size: 12))
                                Text("\(Int(student.distance)) ft")
                                    .font(ColageFonts.captionBold)
                            }
                            .foregroundStyle(ColageColors.textSecondary)
                        }
                    }
                    .padding(.top, 40)

                    // Major
                    if let major = student.profile.major {
                        HStack(spacing: 6) {
                            Image(systemName: "book.fill")
                                .font(.system(size: 14))
                            Text(major)
                                .font(ColageFonts.body)
                        }
                        .foregroundStyle(ColageColors.textSecondary)
                    }

                    // Bio
                    if let bio = student.profile.bio {
                        Text(bio)
                            .font(ColageFonts.body)
                            .foregroundStyle(ColageColors.textPrimary)
                            .multilineTextAlignment(.center)
                            .padding(.horizontal, 32)
                    }

                    // Social links
                    if !student.profile.socialLinks.isEmpty {
                        VStack(spacing: 10) {
                            Text("Connect")
                                .font(ColageFonts.captionBold)
                                .foregroundStyle(ColageColors.textTertiary)
                                .frame(maxWidth: .infinity, alignment: .leading)

                            ForEach(student.profile.socialLinks) { link in
                                Button {
                                    if let url = link.url {
                                        UIApplication.shared.open(url)
                                    }
                                } label: {
                                    HStack(spacing: 14) {
                                        Image(systemName: link.platform.iconName)
                                            .font(.system(size: 18))
                                            .foregroundStyle(ColageColors.primary)
                                            .frame(width: 40, height: 40)
                                            .background(ColageColors.primary.opacity(0.12))
                                            .clipShape(RoundedRectangle(cornerRadius: 10))

                                        VStack(alignment: .leading, spacing: 2) {
                                            Text(link.platform.displayName)
                                                .font(ColageFonts.bodyBold)
                                                .foregroundStyle(ColageColors.textPrimary)
                                            Text(link.handle)
                                                .font(ColageFonts.caption)
                                                .foregroundStyle(ColageColors.textSecondary)
                                        }

                                        Spacer()

                                        Image(systemName: "arrow.up.right")
                                            .font(.system(size: 14))
                                            .foregroundStyle(ColageColors.textTertiary)
                                    }
                                    .padding(12)
                                    .background(ColageColors.surface)
                                    .clipShape(RoundedRectangle(cornerRadius: 14))
                                }
                            }
                        }
                        .padding(.horizontal, 24)
                    }

                    Spacer().frame(height: 40)
                }
            }

            // Close button
            VStack {
                HStack {
                    Spacer()
                    Button {
                        dismiss()
                    } label: {
                        Image(systemName: "xmark")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundStyle(ColageColors.textSecondary)
                            .frame(width: 32, height: 32)
                            .background(ColageColors.surfaceElevated)
                            .clipShape(Circle())
                    }
                    .padding(16)
                }
                Spacer()
            }
        }
    }
}

#Preview {
    FullProfileView(
        student: NearbyStudent.mock(index: 0, baseLat: 42.278, baseLng: -83.738)
    )
}

File: android/app/src/main/java/com/colageclub/colage/features/discovery/FullProfileView.kt
package com.colageclub.colage.features.discovery

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import com.colageclub.colage.core.design.*
import com.colageclub.colage.data.models.NearbyStudent

@Composable
fun FullProfileView(
    student: NearbyStudent,
    onDismiss: () -> Unit
) {
    val context = LocalContext.current

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(ColageColors.Background)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState()),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.height(40.dp))

            // Large avatar
            AvatarView(
                imageUrl = student.profile.profilePhotoURL,
                size = 120.dp,
                initials = student.profile.displayName.initials()
            )

            Spacer(Modifier.height(16.dp))

            Text(
                text = student.profile.displayName,
                style = ColageFonts.Title.copy(color = ColageColors.TextPrimary)
            )

            Spacer(Modifier.height(12.dp))

            // University badge + distance
            Row(
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                // University badge
                Row(
                    modifier = Modifier
                        .background(ColageColors.Primary.copy(alpha = 0.12f), RoundedCornerShape(50))
                        .padding(horizontal = 10.dp, vertical = 5.dp),
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Default.School,
                        contentDescription = null,
                        tint = ColageColors.Primary,
                        modifier = Modifier.size(12.dp)
                    )
                    Text(
                        text = student.profile.universityDomain.removeSuffix(".edu").uppercase(),
                        style = ColageFonts.CaptionBold.copy(color = ColageColors.Primary)
                    )
                }

                // Distance
                Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                    Icon(
                        Icons.Default.LocationOn,
                        contentDescription = null,
                        tint = ColageColors.TextSecondary,
                        modifier = Modifier.size(12.dp)
                    )
                    Text(
                        text = student.distance.formattedDistance(),
                        style = ColageFonts.CaptionBold.copy(color = ColageColors.TextSecondary)
                    )
                }
            }

            // Major
            student.profile.major?.let { major ->
                Spacer(Modifier.height(16.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                    Icon(
                        Icons.Default.MenuBook,
                        contentDescription = null,
                        tint = ColageColors.TextSecondary,
                        modifier = Modifier.size(14.dp)
                    )
                    Text(
                        text = major,
                        style = ColageFonts.Body.copy(color = ColageColors.TextSecondary)
                    )
                }
            }

            // Bio
            student.profile.bio?.takeIf { it.isNotEmpty() }?.let { bio ->
                Spacer(Modifier.height(16.dp))
                Text(
                    text = bio,
                    style = ColageFonts.Body.copy(color = ColageColors.TextPrimary),
                    modifier = Modifier.padding(horizontal = 32.dp)
                )
            }

            // Social links
            if (student.profile.socialLinks.isNotEmpty()) {
                Spacer(Modifier.height(24.dp))
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 24.dp)
                ) {
                    Text(
                        text = "Connect",
                        style = ColageFonts.CaptionBold.copy(color = ColageColors.TextTertiary)
                    )
                    Spacer(Modifier.height(10.dp))
                    student.profile.socialLinks.forEach { link ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 4.dp)
                                .background(ColageColors.Surface, RoundedCornerShape(14.dp))
                                .clickable {
                                    link.url()?.let { url ->
                                        context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
                                    }
                                }
                                .padding(12.dp),
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.spacedBy(14.dp)
                        ) {
                            Box(
                                modifier = Modifier
                                    .size(40.dp)
                                    .background(ColageColors.Primary.copy(alpha = 0.12f), RoundedCornerShape(10.dp)),
                                contentAlignment = Alignment.Center
                            ) {
                                Icon(
                                    link.platform.icon(),
                                    contentDescription = null,
                                    tint = ColageColors.Primary,
                                    modifier = Modifier.size(18.dp)
                                )
                            }
                            Column(modifier = Modifier.weight(1f)) {
                                Text(
                                    text = link.platform.displayName,
                                    style = ColageFonts.BodyBold.copy(color = ColageColors.TextPrimary)
                                )
                                Text(
                                    text = link.handle,
                                    style = ColageFonts.Caption.copy(color = ColageColors.TextSecondary)
                                )
                            }
                            Icon(
                                Icons.Default.OpenInNew,
                                contentDescription = null,
                                tint = ColageColors.TextTertiary,
                                modifier = Modifier.size(14.dp)
                            )
                        }
                    }
                }
            }

            Spacer(Modifier.height(40.dp))
        }

        // Close button
        IconButton(
            onClick = onDismiss,
            modifier = Modifier
                .align(Alignment.TopEnd)
                .padding(16.dp)
                .size(32.dp)
                .clip(CircleShape)
                .background(ColageColors.SurfaceElevated)
        ) {
            Icon(
                Icons.Default.Close,
                contentDescription = "Close",
                tint = ColageColors.TextSecondary,
                modifier = Modifier.size(14.dp)
            )
        }
    }
}
